src/parcsr_ls/par_amg.c

```diff
@@ -514,376 +514,343 @@ HYPRE_Int
 hypre_BoomerAMGDestroy( void *data )
 {
    hypre_ParAMGData  *amg_data = (hypre_ParAMGData*) data;
-   HYPRE_Int num_levels = hypre_ParAMGDataNumLevels(amg_data);
-   HYPRE_Int smooth_num_levels = hypre_ParAMGDataSmoothNumLevels(amg_data);
-   HYPRE_Solver *smoother = hypre_ParAMGDataSmoother(amg_data);
-   void *amg = hypre_ParAMGDataCoarseSolver(amg_data);
-   MPI_Comm new_comm = hypre_ParAMGDataNewComm(amg_data);
-   HYPRE_Int i;
-   HYPRE_Int *grid_relax_type = hypre_ParAMGDataGridRelaxType(amg_data);
 
    HYPRE_ANNOTATE_FUNC_BEGIN;
+   if (amg_data)
```

Comment:
Is it always the case that if amg_data exists then all of the subcomponents do as well?  E.g., hypre_ParAMGDataLambda(amg_data)?

Reply:
Not every AMG component (data structure member) will always exist but when they don't, they are set to `NULL`. E.g.,`hypre_ParAMGDataLambda(amg_data)` seems to be created only when using the Schwarz smoother option, otherwise it's `NULL`. 

Previously we had the check `if (hypre_ParAMGDataLambda(amg_data))` prior to the destroy call:
https://github.com/hypre-space/hypre/blob/922a3ce4df74d3b2b6593ff4d015531f147a8174/src/parcsr_ls/par_amg.c#L646-L649

but that check is not really necessary since `hypre_ParCSRMatrixDestroy` already does that internally, so that's why I removed some of the checks in `hypre_BoomerAMGDestroy`

Reply:
Sounds good.  Thanks!